api/types.proto

```diff
@@ -713,3 +727,31 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+// SecretReference is the linkage between a service and a secret that it uses.
+message SecretReference {
+	// Name is the name of the secret that this references.
+	string name = 1;
+
+	// SecretValueID represents the ID of the specific Secret version that we're
+	// referencing. This identifier exists so that SecretReferences don't leak
+	// any information about the secret contents.
+	string secret_value_id = 2 [(gogoproto.customname) = "SecretValueID"];
```

Comment:
@stevvooe what is an "active" reference? Technically you can rollback to a previous spec, and the previous secret reference becomes active. When do we decide a reference is no longer in use?